datafusion/src/physical_plan/file_format/parquet.rs

```diff
@@ -757,10 +784,8 @@ mod tests {
             .enumerate()
             .map(|(i, g)| row_group_predicate(g, i))
             .collect::<Vec<_>>();
-        // no row group is filtered out because the predicate expression can't be evaluated
-        // when a null array is generated for a statistics column,
-        // because the null values propagate to the end result, making the predicate result undefined
-        assert_eq!(row_group_filter, vec![true, true]);
+        // First row group was filtered out because it contains no null value on "c2".
+        assert_eq!(row_group_filter, vec![false, true]);
```

Comment:
I actually think this could be `vec![false, false]` as the predicate can never be true (`int > 1 AND bool = NULL` is always `NULL`)

Reply:
I am not sure about the expression semantics in datafusion. In Spark, the predicate should be `IsNull` that checks the null value. Here I follow the original expression `bool = NULL`.

I see there is also `IsNull` predicate expression, but I don't see `IsNull` is handled in predicate pushdown. I don't know if this is intentional or a bug.

I can fix it if you agree that `IsNull` is correct way to handle null predicate here.

Reply:
I think this is related to the "Confusion 1".

Reply:
In sql `IsNull` is the correct way to test a column for null as well 👍 

It would make a lot of sense to me to rewrite `x IS NULL` --> `0 > x_null_count`

Reply:
yea, I'm surprised when I looked at the `bool = NULL` and confused too. I guess this is how datafusion works but seems not :). Let me fix it together.

Reply:
Would you like me to fix it here or in a following PR?

Reply:
I've updated to use IsNull for predicate pruning.